neoscore/core/path.py

```diff
@@ -72,17 +72,31 @@ def straight_line(
         start: PointDef,
         parent: Optional[PositionedObject],
         end: PointDef,
+        end_parent: Optional[PositionedObject] = None,
         brush: Optional[BrushDef] = None,
         pen: Optional[PenDef] = None,
     ) -> Path:
         """Convenience for drawing a single straight line.
 
-        ``end`` is measured relative to the starting point.
+        Args:
+            start: The position of the center of the arrow line's start
+            parent: A parent object
+            end: The position of the end of the line, relative to ``end_parent``
+                if provided or ``start``
+            end_parent: An optional parent for the end point.
+            brush: The brush to fill shapes with.
+            pen: The pen to draw outlines with. Defaults to no pen.
+
+        Note that ``end_parent`` is only used for initially drawing the
+        path. If ``end_parent`` moves relative to the path after
+        creation, the path shape will not be automatically updated.
         """
         line = cls(start, parent, brush, pen)
-        if isinstance(end, tuple):
-            end = Point(*end)
-        line.line_to(end.x, end.y)
+        end = Point.from_def(end)
+        if end_parent:
+            line.line_to(end.x, end.y, end_parent)
+        else:
+            line.line_to(end.x, end.y)
```

Comment:
no need to split these calls out, `line_to(x, y, None)` is already the default value if you omit the `end_parent` arg, so you can unconditionally pass `end_parent` in